### PR TITLE
[minor] Mirror openshift release and operators

### PR DESCRIPTION
**Pull request by racree**
Adds two new playbooks and associated roles:
- mirror_openshift_operators
- mirror_openshift_release

### CODE REVIEW

File: ibm/mas_airgap/roles/ocp_operators_mirror/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+redhat_connect_username: "{{ lookup('env', 'REDHAT_CONNECT_USERNAME') }}"
+redhat_connect_password: "{{ lookup('env', 'REDHAT_CONNECT_PASSWORD') }}"
+redhat_connect_auth: "{{ redhat_connect_username }}:{{ redhat_connect_password }}"
+
+quay_username: "{{ lookup('env', 'QUAY_USERNAME') }}"
+quay_password: "{{ lookup('env', 'QUAY_PASSWORD') }}"
+quay_auth: "{{ quay_username }}:{{ quay_password }}"
+
+registry_public_username: "{{ lookup('env', 'REGISTRY_PUBLIC_USERNAME') }}"
```
Comment (durera) [inline ibm/mas_airgap/roles/ocp_operators_mirror/defaults/main.yml:22]: Can we keep the naming of these vars consistent with existing roles: `registry_auth_username` and `registry_auth_password`

Comment (durera) [inline ibm/mas_airgap/roles/ocp_operators_mirror/defaults/main.yml:22]: Can we keep the naming of these vars consistent with existing roles: `registry_auth_username` and `registry_auth_password`

File: ibm/mas_airgap/roles/ocp_operators_mirror/README.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
+case_mirror
```
Comment (durera) [inline ibm/mas_airgap/roles/ocp_operators_mirror/README.md:1]: README needs to be updated, it's a clone of another role ...

Comment (durera) [inline ibm/mas_airgap/roles/ocp_operators_mirror/README.md:1]: README needs to be updated, it's a clone of another role ...

File: ibm/mas_airgap/roles/ocp_release_mirror/README.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
+case_mirror
```
Comment (durera) [inline ibm/mas_airgap/roles/ocp_release_mirror/README.md:1]: README needs to be updated, it's a clone of another role

Comment (durera) [inline ibm/mas_airgap/roles/ocp_release_mirror/README.md:1]: README needs to be updated, it's a clone of another role

File: ibm/mas_airgap/roles/ocp_release_mirror/defaults/main.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
+---
+# Get Registry facts from environment variables if they have not been passed to the role
+registry_public_host: "{{ lookup('env', 'REGISTRY_PUBLIC_HOST') }}"
+registry_public_port: "{{ lookup('env', 'REGISTRY_PUBLIC_PORT') }}"
+registry_public_url: "{{ registry_public_host }}:{{ registry_public_port }}"
+
+# Release config
+openshift_release_version: "{{ lookup('env', 'OPENSHIFT_RELEASE_VERSION') }}"
+dir: "{{ lookup('env', 'DIR') }}"
```
Comment (durera) [inline ibm/mas_airgap/roles/ocp_release_mirror/defaults/main.yml:9]: As before, can we use `log_dir` so purpose is clearer.

Comment (durera) [inline ibm/mas_airgap/roles/ocp_release_mirror/defaults/main.yml:9]: As before, can we use `log_dir` so purpose is clearer.

File: ibm/mas_airgap/roles/ocp_release_mirror/defaults/main.yml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+ibm_entitlement_key: "{{ lookup('env', 'IBM_ENTITLEMENT_KEY') }}"
```
Comment (durera) [inline ibm/mas_airgap/roles/ocp_release_mirror/defaults/main.yml:11]: Similar ... I don't see either ibm_entitlement_key or ibm_auth being used in this role, so think we can remove them from here

Comment (durera) [inline ibm/mas_airgap/roles/ocp_release_mirror/defaults/main.yml:11]: Similar ... I don't see either ibm_entitlement_key or ibm_auth being used in this role, so think we can remove them from here

File: ibm/mas_airgap/roles/ocp_release_mirror/defaults/main.yml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+redhat_connect_username: "{{ lookup('env', 'REDHAT_CONNECT_USERNAME') }}"
+redhat_connect_password: "{{ lookup('env', 'REDHAT_CONNECT_PASSWORD') }}"
+redhat_connect_auth: "{{ redhat_connect_username }}:{{ redhat_connect_password }}"
+
+quay_username: "{{ lookup('env', 'QUAY_USERNAME') }}"
+quay_password: "{{ lookup('env', 'QUAY_PASSWORD') }}"
+quay_auth: "{{ quay_username }}:{{ quay_password }}"
+
+registry_public_username: "{{ lookup('env', 'REGISTRY_PUBLIC_USERNAME') }}"
```
Comment (durera) [inline ibm/mas_airgap/roles/ocp_release_mirror/defaults/main.yml:22]: Align the name with the existing registry_auth_username and registry_auth_password properties

Comment (durera) [inline ibm/mas_airgap/roles/ocp_release_mirror/defaults/main.yml:22]: Align the name with the existing registry_auth_username and registry_auth_password properties

**Review by durera — CHANGES_REQUESTED** [review]

File: ibm/mas_airgap/roles/ocp_operators_mirror/defaults/main.yml [hunk 7]
```diff
@@ -0,0 +1,25 @@
+---
+# Get Registry facts from environment variables if they have not been passed to the role
+registry_public_host: "{{ lookup('env', 'REGISTRY_PUBLIC_HOST') }}"
+registry_public_port: "{{ lookup('env', 'REGISTRY_PUBLIC_PORT') }}"
+registry_public_url: "{{ registry_public_host }}:{{ registry_public_port }}"
+
+# Release config
+openshift_operators_version: "{{ lookup('env', 'OPENSHIFT_OPERATORS_VERSION') }}"
+dir: "{{ lookup('env', 'DIR') }}"
```
[the remark this answers is not in the document]
Comment (racree) [reply ibm/mas_airgap/roles/ocp_operators_mirror/defaults/main.yml:9]: ok.. yes, I'll rename it.

File: ibm/mas_airgap/roles/ocp_operators_mirror/defaults/main.yml [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+ibm_entitlement_key: "{{ lookup('env', 'IBM_ENTITLEMENT_KEY') }}"
```
[the remark this answers is not in the document]
Comment (racree) [reply ibm/mas_airgap/roles/ocp_operators_mirror/defaults/main.yml:11]: ibm_entitlement_key is used to create ibm_auth on line 12 of defaults/main.yml

File: ibm/mas_airgap/roles/ocp_operators_mirror/defaults/main.yml (hunk 1 above)
Comment (racree) [reply ibm/mas_airgap/roles/ocp_operators_mirror/defaults/main.yml:22]: yeah I agree.. in fact I saw many references to registry_username and registry_password and prefer this naming.. Is that ok?

File: ibm/mas_airgap/roles/ocp_release_mirror/defaults/main.yml (hunk 4 above)
Comment (racree) [reply ibm/mas_airgap/roles/ocp_release_mirror/defaults/main.yml:9]: ok

File: ibm/mas_airgap/roles/ocp_release_mirror/defaults/main.yml (hunk 5 above)
Comment (racree) [reply ibm/mas_airgap/roles/ocp_release_mirror/defaults/main.yml:11]: ibm_entitlement_key is used to craete ibm_auth.  ibm_auth is used on line 69 of ocp_release_mirror/tasks/main.yml

Comment (racree) [reply ibm/mas_airgap/roles/ocp_release_mirror/defaults/main.yml:11]: ibm_entitlement_key is used to create ibm_auth. ibm_auth is used on line 69 of ocp_release_mirror/tasks/main.yml

File: ibm/mas_airgap/roles/ocp_release_mirror/defaults/main.yml (hunk 6 above)
Comment (racree) [reply ibm/mas_airgap/roles/ocp_release_mirror/defaults/main.yml:22]: I updated all the references to use "registry_username and registry_password".  To me this is more consistent with the rest of the ansible scripts.  Please let me know if this is ok.

**Review by durera — DISMISSED** [review]

**Review by durera — APPROVED** [review]

**Pull request merged**